tmt/steps/prepare/feature/crb.py

```diff
@@ -0,0 +1,84 @@
+import re
+from typing import Any, Optional
+
+import tmt.log
+from tmt.container import container, field
+from tmt.package_managers import Package
+from tmt.steps.prepare.feature import PrepareFeatureData, ToggleableFeature, provides_feature
+from tmt.steps.provision import Guest
+from tmt.utils import ShellScript
+
+SUPPORTED_DISTRO_PATTERNS = (
+    re.compile(pattern)
+    for pattern in (r'Red Hat Enterprise Linux (8|9|10)', r'CentOS Stream (8|9|10)')
+)
+
+
+@container
+class CrbStepData(PrepareFeatureData):
+    crb: Optional[str] = field(
+        default=None,
+        option='--crb',
+        metavar='enabled|disabled',
+        help='Whether the CRB repository should be enabled or disabled.',
+    )
+
+
+@provides_feature('crb')
+class Crb(ToggleableFeature):
+    """
+    Enable or disable the CodeReady Builder (CRB) repository:
+
+    .. code-block:: yaml
+
+        prepare:
+            how: feature
+            crb: enabled
+
+    .. code-block:: shell
+
+        prepare --how feature --crb enabled
+
+    .. versionadded:: 1.49
+    """
+
+    _data_class = CrbStepData
+
+    def __init__(self, *args: Any, **kwargs: Any) -> None:
+        super().__init__(*args, **kwargs)
+
+    @classmethod
+    def _check_distro(cls, guest: Guest, logger: tmt.log.Logger) -> None:
+        """Verify the guest distribution is supported and install config-manager if needed"""
+        if not (
+            guest.facts.distro
+            and any(pattern.match(guest.facts.distro) for pattern in SUPPORTED_DISTRO_PATTERNS)
+        ):
+            logger.warning('CRB prepare feature is supported on RHEL/CentOS-Stream 8, 9 or 10.')
```

Comment:
Emits a warning, but then proceeds and tries to make changes to guest that is reportedly not compatible. Since `_check_distro()` is called just once, would it make sense for its code to be merged into `_manage_repo()`, and `warning()` followed by `return`?

Reply:
Or return `bool` denoting if the check was successful and use it for the `return` decision:

```python
if not cls._check_distro(guest, logger):
    return
```

Incorporating sounds also ok, it's not that large.

Reply:
Should be covered by cb01e7ab.